conu/utils/__init__.py

```diff
@@ -109,3 +114,82 @@ def mkdtemp():
 def random_tmp_filename():
     """ generate string which can be used as a filename for temporary file """
     return "conu-" + random_str(32)
+
+
+class CommandDoesNotExistException(ConuException):
+    """ Requested command is not present on the system """
+
+
+def command_exists(command, noop_invocation, exc_msg):
+    """
+    Verify that the provided command exists.
+
+    :param command: str, command to check (python 3 only)
+    :param noop_invocation: list of str, command to check (python 2 only)
+    :param exc_msg: str, message of exception when command does not exist
+    :return: None
+    """
+    try:
+        found = bool(shutil.which(command))  # py3 only
+    except AttributeError:  # py2 branch
+        try:
+            p = subprocess.Popen(noop_invocation, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        except OSError:
+            found = False
+        else:
+            stdout, stderr = p.communicate()
+            found = p.returncode == 0
+            if not found:
+                logger.error("`%s` exited with a non-zero return code (%s)",
+                             noop_invocation, p.returncode)
+                logger.error("command stdout = %s", stdout)
+                logger.error("command stderr = %s", stderr)
+    if not found:
+        raise CommandDoesNotExistException(exc_msg)
+
```

Comment:
We could add `return True`, so the 'question' in method header is answered - in case someone writes sth like
`assert s2i_command_exists()` as I did and then I got a bit confused :)

Reply:
I originally had it like that but changed it last second; will do